lib/services/local/users.go

```diff
@@ -563,12 +563,15 @@ func (s *IdentityService) CompareAndSwapUser(ctx context.Context, new, existing
 
 		item.Revision = currentWithoutSecrets.GetRevision()
 
-		if _, err = s.Backend.ConditionalUpdate(ctx, item); err != nil {
+		lease, err := s.Backend.ConditionalUpdate(ctx, item)
+		if err != nil {
 			if trace.IsCompareFailed(err) {
 				continue
 			}
 			return trace.Wrap(err)
 		}
+		new.SetRevision(lease.Revision)
+		new.SetWeakestDevice(newWithoutSecrets.GetWeakestDevice())
```

Comment:
Is this a bugfix?

Reply:
Are we relying on this to update `new` for the caller's sake? I don't see how updating `new` has any effect other than that, we're going to call `upsertLocalSecrets` with `new.GetName()` and a shallow copy of `new.GetLocalAuth()` and then we'll return from this.

Reply:
Is it just broken tests that rely on inspecting the value of the "new" item after `CompareAndSwap` returns, or is it actual production code that does? Either way, it's a really bad sign.